src/inverted-index.js

```diff
@@ -0,0 +1,116 @@
+class InvertedIndex{
+	
+	constructor() {
+		/**
+		 * Initialize an empty array to hold the
+		 * concantenated words formed from title and text
+		 * contents.
+		 */ 
+		this.indexWords = [];
+		/**
+		 * Initialize an empty object to hold the mapping
+		 * of the words to their file location 
+		 */
+		this.index = {};
+		/**
+		 * Initialize an empty array to hold the 
+		 * search results for words
+		 */
+		this.searchResults = [];
+	}
+	
+	/**
+	 * This Method returns an array of tokens/words from the provided input
+	 * after removing special characters and white spaces
+	 * @params{String} str - String to be tokonized
+	 */
+	tokenize(str) {
+	    var cleanString = str.trim().replace(/-/g, " ").replace(/[.,\/#!$%\^&@\*;:'{}=\_`~()]/g, "").toLowerCase().split(" ").sort();
+	    return cleanString;
+	}
+	
+	/**
+	 * This method returns an array of tokenized unique words
+	 * of the provided input 
+	 * @params{String} str - The string to be filtered
+	 */
+	uniqueWords(str) {
+		var tokens = this.tokenize(str);
+		return tokens.filter(function (item, pos) {
+      		return tokens.indexOf(item) === pos;
```

Comment:
A little baffling that i just pulled down this branch into a local repo and this portion of the code is showing that its properly indented( line 40 properly indented under 39). Perhaps its a tab and Space mix; will try uncover the reason and fix.